core/blockchain.go

```diff
@@ -1796,6 +1836,10 @@ func (bc *BlockChain) insertChain(chain types.Blocks, setHead bool) (int, error)
 		dirty, _ := bc.triedb.Size()
 		stats.report(chain, it.index, dirty, setHead)
 
+		if bc.logger != nil {
+			bc.logger.OnBlockEnd(nil)
```

Comment:
It's a good idea. I pushed a commit but as you can see the diff blew up because this code is in a for loop. So I had to wrap the relevant part of block import into a function to be able to use defer.

https://github.com/s1na/go-ethereum/commit/abd880712babb5c251c4d4cb31ea7aed26b4c47c